pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -1505,6 +1531,53 @@ func (r *ReconcileClusterDeployment) cleanupInstallLogPVC(cd *hivev1.ClusterDepl
 	return nil
 }
 
+// deleteInstallPods deletes the install pods for the given provision
+func (r *ReconcileClusterDeployment) deleteInstallPods(provision *hivev1.ClusterProvision, pLog log.FieldLogger) error {
```

Comment:
The controller should not be deleting the install pod directly. It should be deleting the install job, which will cascade the delete to the install pod. If this is done by the clusterprovision controller, then there is no need to search for the job: The ClusterProvision has a reference to the job in its status.